### DISCUSSION TITLE
Rebuild horovod with customer nccl library

### DISCUSSION THREAD

**Discussion by JoeyYoung** [horovod/horovod#3232] [Q&A] [answered]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.9.1
3. Horovod version: v0.23.0
5. CUDA version: 11.2
6. NCCL version: 2.11.4
7. Python version: 3.9

Hello, 
I have installed Horovod with a customer nccl library. But everytime I modify the nccl library, I need to rebuild horovod with following steps to make sure it uses modified nccl:

```
cd ~/horovod
python setup.py sdict
HOROVOD_NCCL_HOME=/path/to/nccl HOROVOD_GPU_OPERATIONS=NCCL pip install --no-cache-dir dist/horovod-0.23.0.tar.gz
```

Is this a must? Any suggestion to simplify the rebuild process?

**Comment by maxhgerlach** [comment, marked as answer]
Hi @JoeyYoung,

check out the environment variables that you can set to tweak the build: https://horovod.readthedocs.io/en/stable/install_include.html#environment-variables

You should be able to set `HOROVOD_NCCL_LINK=SHARED` to link dynamically against NCCL.

**Reply by JoeyYoung** [reply]
I see. Thanks!

**Reply by maxhgerlach** [reply]
Good luck!

... I fixed a stupid typo in the answer.
